pkg/actions/lua/lakefs/catalogexport/hive.lua

```diff
@@ -0,0 +1,68 @@
+local pathlib = require("path")
+local utils = require("lakefs/catalogexport/internal/utils")
+
+-- extract partition prefix from full path
+function extract_partitions_path(partitions, path)
+    local idx = 0
+    for _, partition in ipairs(partitions) do
+        local col_substr = partition .. "="
+        local i, j = string.find(path, col_substr, idx)
+        if i == nil then
+            return nil
+        end
+        local start_val, end_val = string.find(path, "/", j+1)
+        idx = end_val 
+    end
+    return string.sub(path, 1, idx)
+end
+
+-- Hive format partition iterator each result set is a collection of files under the same partition
+function lakefs_hive_partition_pager(client, repo_id, commit_id, base_path, page_size, partition_cols)
```

Comment:
1. its already re-ordered, you're not rebased 
2. i prefer to keep the prefix explicit since there are going to be iterations around tables and objects in the export destination as well.